plugins/modules/cloud/vultr/vultr_lb.py

```diff
@@ -0,0 +1,307 @@
+
+        if len(found_lbs) in (0, 1):
+            self.module.warn("Found more than 1 or no Vultr Load Balancer matching {}".format(lb_name))
+            return {}
+
+        return found_lbs[0]
+
+    def present_lb(self):
+        lb = self.get_lb()
+        if not lb:
+            lb = self._create_lb(lb)
+        return lb
+
+    def _create_lb(self, lb):
+        self.result['changed'] = True
+        data = {
+            'DCID': self.module.params.get('dcid'),
+            'label': self.module.params.get('name'),
+        }
+
+        def add_param_if_exists(d, param):
+            value = self.module.params.get(param)
+            if value:
+                d[param] = value
+
+        optional_params = [
+            'config_ssl_redirect',
+            'sticky_sessions',
+            'cookie_name',
+            'balancing_algorithm',
+            'health_check',
+            'forwarding_rules',
+            'ssl_private_key',
+            'ssl_certificate',
+            'ssl_chain'
+        ]
+
+        for param in optional_params:
+            add_param_if_exists(data, param)
+
+        # StickSessions should be either 'on' or 'off'
+        if 'sticky_sessions' in data:
+            ss_values = { True: 'on', False: 'off'}
+            data['sticky_sessions'] = ss_values[data['sticky_sessions']]
+
+        self.result['diff']['before'] = {}
+        self.result['diff']['after'] = data
+
+        if not self.module.check_mode:
+            self.api_query(
+                path="/v1/loadbalancer/create",
+                method="POST",
+                data=data
+            )
+            lb = self.get_lb()
+        return lb
+
+    def absent_lb(self):
+        lb = self.get_lb()
+        if lb:
+            self.result['changed'] = True
+
+        data = {
+            'label': lb['name'],
+        }
+
+        self.result['diff']['before'] = lb
+        self.result['diff']['after'] = {}
+
+        if not self.module.check_mode:
+            self.api_query(
+                path="/v1/loadbalancer/create",
+                method="POST",
+                data=data
+            )
+        return lb
+
+
+def main():
+    argument_spec = vultr_argument_spec()
+    argument_spec.update({
+        'name': {
+            'required': True,
+            'aliases': ['domain'],
+        },
+        'dcid': {
+            'required': True,
+        },
+        'config_ssl_redirect': {},
+        'sticky_sessions': {},
+        'cookie_name': {},
+        'balancing_algorithm': {},
+        'health_check': {},
+        'forwarding_rules': {},
+        'ssl_private_key': {},
+        'ssl_certificate': {},
+        'ssl_chain': {},
+        'state': {
+            'choices': ['present', 'absent'],
+            'default': 'present',
+        },
+    })
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        required_if=[
+            ('state', 'present', ['name', 'dcid']),
```

Comment:
This doesn't make sense, since you marked both options `name` and `dcid` as always required.